application/src/main/java/net/corda/v5/application/interop/binding/InteropAction.java

```diff
@@ -0,0 +1,36 @@
+package net.corda.v5.application.interop.binding;
+
+import net.corda.v5.application.interop.facade.FacadeRequest;
+
+public abstract class InteropAction<T> {
+    public abstract T getResult();
+
+    public static final class ClientAction<T> extends InteropAction<T> {
```

Comment:
I'm unfamiliar with this pattern of nested classes extending (publicly) their nesting abstract class.
Can you explain why not declare them outside `class InteropAction<T>`

Reply:
This type of nesting is purely for a namespace solution, `InteropAction` acts like package name for `ClientAction`.
The rest is as the usual concrete class `ClientAction` implements abstract class `InteropAction`.